sceptre/resolvers/stack_output.py

```diff
@@ -163,21 +173,43 @@ def resolve(self):
         """
         self.logger.debug("Resolving external Stack output: {0}".format(self.argument))
 
-        profile = None
-        region = None
-        sceptre_role = None
         arguments = shlex.split(self.argument)
 
+        if not arguments:
+            message = "!stack_output_external requires at least one argument"
+            raise SceptreException(message)
+
         stack_argument = arguments[0]
+        stack_args = iter(stack_argument.split("::"))
+
+        try:
+            dependency_stack_name = next(stack_args)
+            output_key = next(stack_args)
+
+        except StopIteration as err:
+            message = "!stack_output_external arg should match STACK_NAME::OUTPUT_KEY"
+            raise SceptreException(message) from err
+
+        profile = region = sceptre_role = None
+
         if len(arguments) > 1:
```

Comment:
I don't know why you went this route with the condition here. Why the distinction between the `arguments` and the `extra args`? 

Why not just effectively do:

```python
args = iter(argument.split(":::"))
dependency_stack_name = next(args)
output_key = next(args)
profile = next(args, None)
region = next(args, None)
sceptre_role = next(args, None)
```

Reply:
Something like this?

<img width="802" alt="Screenshot 2023-12-15 at 12 00 15 am" src="https://github.com/Sceptre/sceptre/assets/2305330/700bca5a-adce-4ae4-82d3-657da6ad89b3">

Reply:
@jfalkenstein On testing your idea (at least as I understood it), I realised why the `if` conditional in the original code is there. We can't just do this:
```python
args = iter(argument.split("::"))
```
Because the resolver receives an argument of the form:

```
STACK_NAME::OUTPUT_KEY[ PROFILE[::REGION[::SCEPTRE_ROLE]]]
```

So we need to firstly split on the space (the `if` condition in effect does that) and only then split the 2 parts on `::`.

I have therefore reverted to the same version that reviewed last time.

Could you have another look?